src/interpreter/mod.rs

```diff
@@ -1404,6 +1404,10 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
             (8, &ty::TyUint(UintTy::Us)) |
             (_, &ty::TyUint(UintTy::U64)) => PrimVal::U64(self.memory.read_uint(ptr, 8)? as u64),
 
+            (_, &ty::TyFnDef(def_id, substs, fn_ty)) => {
+                PrimVal::FnPtr(self.memory.create_fn_ptr(def_id, substs, fn_ty))
```

Comment:
I wonder if we need to differentiate between `FnPtr` and `FnDef` primvals. `Memory::write_primval` might need to know which one it's writing (so it doesn't write for a `FnDef`). I don't know if this can happen yet, but perhaps in the future?